### ISSUE TITLE
Hide page menu (top CollectionView) when there's only single page

### ISSUE THREAD

**Issue by Maxlufs** [rechsteiner/Parchment#378]
Is there anyway to hide the page menu (on the top) when there's only one page?
something like UIPageControl
`hidesForSinglePage = true`

seems PageView sets height constraint on page menu (the CollectionView) and i couldn't find a way to override that heightAnchor except subclassing PagingView and override the entire setupConstraints()

**Comment by rechsteiner** [comment]
Hi @Maxlufs! There isn't any build in way of doing it, so you would have to create a `PagingView` subclass like you mentioned 🙂

**Comment by Pratik948** [comment]
Hi @Maxlufs ! I have done similar with my project, you'll have to set height of menu item to `0` when there's only one.
like this

`pagingViewController.menuItemSize = .sizeToFit(minWidth: 120, height: self.items.count == 1 ? 0 : 40)`

(@rechsteiner I think there's a built in way to do it like above code, I don't know if its a proper way or not)

**Comment by Maxlufs** [comment]
@Pratik948 Thanks! `menuItemSize` works on initialization, but once it's set, I have failed to find other ways to update it. For example: re-setting `menuItemSize` wouldn't work in `viewWillAppear()`
What i'm looking for is basically `hidesForSinglePage`, a sample use case is dependent on if a user is logged in. If they are, show more tabs, if they are logged out, show one.

**Comment by Maxlufs** [comment]
@rechsteiner would be really nice to have that feature if it's not to much of a change 😆

**Comment by rechsteiner** [comment]
@Maxlufs I don't think this will be added as a built-in option to Parchment. It's not that it requires a lot of change, but Parchment would get quite bloated if we added configuration options for every specific use-case. The goal with Parchment is to keep it as generic as possible and rather provide the customisation hooks you need to create your own behaviour.

That said, I'll look into making the menu height update if you change `menuItemSize` after the first render.

**Comment by nathanwhy** [comment]
can we hide page menu for all the pages? `menuItemSize ` may not be a good way.

**Comment by crabman448** [comment]
Would love to see it built into Parchment 😍
Although I achieved desired behaviour it seems to me that subclassing `PagingView` and `PagingViewController` is an overkill for achieving such kinda small thing. 

What I did in my project and what could be also implemented in Parchment from my point of view:
```
class CalendarPagingView: PagingView {
    lazy var menuHeightConstraint: NSLayoutConstraint = {
        collectionView.heightAnchor.constraint(equalToConstant: options.menuHeight)
    }()

    var menuIsHidden: Bool = false {
        didSet {
            menuHeightConstraint.constant = menuIsHidden ? 0.0 : options.menuHeight
        }
    }
    
    // Other code
    
}

class CalendarPagingViewController: PagingViewController<CalendarPagingItem> {
    // In Parchment we don't need this as we have private pagingView
    var viewAsPagingView: CalendarPagingView {
        return view as! CalendarPagingView
    }

    var menuIsHidden: Bool {
        get { return viewAsPagingView.menuIsHidden }
        set { viewAsPagingView.menuIsHidden = newValue }
    }
    
    // Other code
    
}
```

Thank you!

**Comment by rechsteiner** [comment]
Fixed it so that the menu height constraint is now updating when setting `menuItemSize` after the initial render: #449. This should allow you to hide the menu for single items 🙂 I don't really see the need for a custom property for that.

You have to target the `master` branch for now. Will release a new version soon

**Comment by rechsteiner** [comment]
The new version is now released (v2.2.0) 🙂
